src/validator/certvalidators/SubjectCertificatePurposeValidator.php

```diff
@@ -44,22 +47,33 @@ public function __construct(LoggerInterface $logger = null)
     /**
      * Validates that the purpose of the user certificate from the authentication token contains client authentication.
      *
-     * @copyright 2022 Petr Muzikant pmuzikant@email.cz
-     * 
-     * @param subjectCertificate user certificate to be validated
+     * @param X509 $subjectCertificate user certificate to be validated
      * @throws UserCertificateMissingPurposeException
+     * @throws UserCertificateWrongPurposeException
+     * @copyright 2022 Petr Muzikant pmuzikant@email.cz
+     *
      */
     public function validate(X509 $subjectCertificate): void
     {
-        $usages = $subjectCertificate->getExtension('id-ce-extKeyUsage');
-        if (!$usages || empty($usages)) {
+        $keyUsage = $subjectCertificate->getExtension(self::KEY_USAGE);
+        if (!$keyUsage || empty($keyUsage)) {
             throw new UserCertificateMissingPurposeException();
         }
+        if (!$keyUsage[self::KEY_USAGE_DIGITAL_SIGNATURE]) {
+            throw new UserCertificateWrongPurposeException();
+        }
+        $usages = $subjectCertificate->getExtension(self::EXTENDED_KEY_USAGE);
+        if (!$usages || empty($usages)) {
+            // Digital Signature extension present, but Extended Key Usage extension not present,
+            // assume it is an authentication certificate (e.g. Luxembourg eID).
+            return;
```

Comment:
Line 76 has a debug log message, it is needed here as well before return for consistency.